src/formElementPlugins/number/src/ApplicationView.tsx

```diff
@@ -162,18 +196,32 @@ const parseInput = (
   textInput: string | null | undefined,
   numberFormatter: Intl.NumberFormat,
   simple: boolean,
-  prefix: string,
-  suffix: string
-): [number | null, string | null] => {
-  if (!textInput) return [null, null]
+  prefix: string | undefined,
+  suffix: string | undefined,
+  suffixPlural: string | undefined
+): {
+  number: number | null
+  formattedNumber: string | null
+  currentSuffix: string | undefined
+  fullText: string | null
+} => {
```

Comment:
Was starting to get unwieldy, so returning named object fields easier than array.

Quick explanation:
- `number`: the actual number (as JS number type)
- `formattedNumber`: stringified version of number, formatted according to locale/currency/simple settings (e.g. 1,000, $100.00)
- `currentSuffix`: which suffix to use (plural or singular) depending on the number value
- `fullText`: "formattedNumber" plus any prefix or suffix values (e.g. "12 months") -- this is the value that gets stored in the "text" field of the saved response